contracts/PetPark.sol

```diff
@@ -1,7 +1,108 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
+    address owner;
+
+    enum AnimalType {
+        None,
+        Fish,
+        Cat,
+        Dog,
+        Rabbit,
+        Parrot
+    }
+    enum Gender {
+        Male,
+        Female
+    }
+
+    struct Borrower {
+        uint age;
+        Gender gender;
+    }
+
+    mapping(address => AnimalType) borrowedAnimals;
+    mapping(address => Borrower) borrowerList;
+    mapping(AnimalType => uint) public animalCounts;
+
+    event Added(AnimalType animal, uint animalCount);
+    event Borrowed(AnimalType animal);
+    event Returned(AnimalType animal);
+
+    modifier validAnimalType(AnimalType animal) {
+        require(
+            animal > AnimalType.None && animal <= AnimalType.Parrot,
+            "Invalid animal type"
+        );
+        _;
+    }
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    // give shelter to animals in the park
+    function add(AnimalType animal, uint count) external {
+        require(msg.sender == owner, "Not an owner");
```

Comment:
Better to use a modifier for this check since it provides better code readability